lib/pub_sub/tasks/pub_sub.rake

```diff
@@ -17,13 +16,13 @@ namespace :pub_sub do
   def start_poll_thread
     PubSub::Poller.new.poll
   rescue => e
-    NewRelic::Agent.notice_error(e) if defined?(NewRelic)
-    PubSub.logger.error("Unknown error polling subscriptions: #{e.inspect}\n#{e.backtrace}")
+    # Unexpected error during poll. Report and retry.
+    PubSub.report_error(e)
     retry
   end
 
   # How many threads to use for workers
   def worker_concurrency
-    ENV.fetch('PUB_SUB_WORKER_CONCURRENCY', 2).to_i
+    ENV.fetch('PUB_SUB_WORKER_CONCURRENCY', 4).to_i
```

Comment:
I guess I was wondering how we came up with `4`, I am guessing we just decided to double it?